tests/app/main/test_briefs.py

```diff
@@ -2402,19 +1912,107 @@ def test_supplier_details_shown(self, data_api_client):
 
-        assert res.status_code == 200
-        doc = html.fromstring(res.get_data(as_text=True))
-        assert doc.xpath('//h1')[0].text.strip() == "Your application for ‘I need a thing to do a thing’"
-        assert doc.xpath('//p[contains(@class, "banner-message")]')[0].text.strip() == \
-            "You’ve already applied so you can’t apply again."
+        render_template.assert_called_with(
+            "briefs/not_is_supplier_eligible_for_brief_error.html",
+            clarification_question=True,
+            framework_name='Digital Outcomes and Specialists',
+            lot=mock.ANY,
+            reason=mock.ANY,
+            data_reason_slug=mock.ANY,
+        )
+
+    def test_not_on_framework(self, render_template, current_user, data_api_client):
+        current_user.supplier_id = 100
+        data_api_client.find_services.return_value = {"services": []}
+
+        _render_not_eligible_for_brief_error_page(self.brief)
+
+        data_api_client.find_services.assert_called_once_with(
+            supplier_id=100,
+            framework='digital-outcomes-and-specialists',
+            status='published'
+        )
+        render_template.assert_called_with(
+            "briefs/not_is_supplier_eligible_for_brief_error.html",
+            clarification_question=False,
+            framework_name='Digital Outcomes and Specialists',
+            lot='digital-specialists',
+            reason='supplier-not-on-framework',
+            data_reason_slug='supplier-not-on-digital-outcomes-and-specialists',
+        )
+
+    def test_not_on_lot(self, render_template, current_user, data_api_client):
+        current_user.supplier_id = 100
+        data_api_client.find_services.side_effect = lambda *args, **kwargs: (
+            {"services": [{"something": "nonempty"}]} if kwargs.get("lot") is None else {"services": []}
+        )
+
+        _render_not_eligible_for_brief_error_page(self.brief)
+
+        data_api_client.find_services.assert_has_calls(
+            [
+                mock.call(
+                    supplier_id=100,
+                    framework='digital-outcomes-and-specialists',
+                    status='published'
+                ),
+                mock.call(
+                    supplier_id=100,
+                    framework='digital-outcomes-and-specialists',
+                    status='published',
+                    lot='digital-specialists'
+                ),
+            ],
+            any_order=True
+        )
+        render_template.assert_called_with(
+            "briefs/not_is_supplier_eligible_for_brief_error.html",
+            clarification_question=False,
+            framework_name='Digital Outcomes and Specialists',
+            lot='digital-specialists',
+            reason='supplier-not-on-lot',
+            data_reason_slug='supplier-not-on-lot',
+        )
+
+    def test_not_on_role(self, render_template, current_user, data_api_client):
```

Comment:
We can change it. I believe your comment was about the fact it could be clearer. Maybe I'll do something more like `test_supplier_has_service_on_lot_but_not_for_this_role` or `test_supplier_does_not_have_service_with_correct_role`. Thoughts?